Tests/SwiftGRPCTests/ChannelConnectivityTests.swift

```diff
@@ -46,4 +47,21 @@ extension ChannelConnectivityTests {
 
     waitForExpectations(timeout: 0.5)
   }
+
+  func testMultipleConnectivityObserversAreCalled() {
+    let completionHandlerExpectation = expectation(description: "completion handler called")
+    var firstObserverCalled = false
+    var secondObserverCalled = false
+
+    client.channel.addConnectivityObserver { _ in firstObserverCalled = true }
```

Comment:
How about using expectations for these two as well?

Reply:
I had done that initially, but these actually get called multiple times since the state changes more than once. Last I checked, Linux didn't support `assertForOverFulfill` or `expectedFulfillmentCount`, so I think we're stuck with this since CI doesn't use macOS. More info here: https://bugs.swift.org/browse/SR-6249

Reply:
Ah, that makes sense. Would you mind adding a comment explaining this?

Reply:
Sure